packages/aws-cdk-lib/aws-elasticloadbalancingv2/test/alb/load-balancer.test.ts

```diff
@@ -645,6 +645,344 @@ describe('tests', () => {
+        bucket,
+      });
+
+      // WHEN
+      lb.logConnectionLogs(bucket);
+
+      // THEN
+      // verify that the LB attributes reference the bucket
+      Template.fromStack(stack).hasResourceProperties('AWS::ElasticLoadBalancingV2::LoadBalancer', {
+        LoadBalancerAttributes: Match.arrayWith([
+          {
+            Key: 'connection_logs.s3.enabled',
+            Value: 'true',
+          },
+          {
+            Key: 'connection_logs.s3.bucket',
+            Value: 'imported-bucket',
+          },
+          {
+            Key: 'connection_logs.s3.prefix',
+            Value: '',
+          },
+        ]),
+      });
+
+      // verify the bucket policy allows the ALB to put objects in the bucket
+      Template.fromStack(stack).hasResourceProperties('AWS::S3::BucketPolicy', {
+        PolicyDocument: {
+          Version: '2012-10-17',
+          Statement: [
+            {
+              Action: 's3:PutObject',
+              Effect: 'Allow',
+              Principal: { AWS: { 'Fn::Join': ['', ['arn:', { Ref: 'AWS::Partition' }, ':iam::127311923021:root']] } },
+              Resource: {
+                'Fn::Join': [
+                  '',
+                  [
+                    'arn:',
+                    { Ref: 'AWS::Partition' },
+                    ':s3:::imported-bucket/AWSLogs/',
+                    { Ref: 'AWS::AccountId' },
+                    '/*',
+                  ],
+                ],
+              },
+            },
+            {
+              Action: 's3:PutObject',
+              Effect: 'Allow',
+              Principal: { Service: 'delivery.logs.amazonaws.com' },
+              Resource: {
+                'Fn::Join': [
+                  '',
+                  [
+                    'arn:',
+                    { Ref: 'AWS::Partition' },
+                    ':s3:::imported-bucket/AWSLogs/',
+                    { Ref: 'AWS::AccountId' },
+                    '/*',
+                  ],
+                ],
+              },
+              Condition: { StringEquals: { 's3:x-amz-acl': 'bucket-owner-full-control' } },
+            },
+            {
+              Action: 's3:GetBucketAcl',
+              Effect: 'Allow',
+              Principal: { Service: 'delivery.logs.amazonaws.com' },
+              Resource: {
+                'Fn::Join': [
+                  '',
+                  [
+                    'arn:',
+                    { Ref: 'AWS::Partition' },
+                    ':s3:::imported-bucket',
+                  ],
+                ],
+              },
+            },
+          ],
+        },
+      });
+
+      // verify the ALB depends on the bucket policy
+      Template.fromStack(stack).hasResource('AWS::ElasticLoadBalancingV2::LoadBalancer', {
+        DependsOn: ['ImportedConnectionLoggingBucketPolicy548EEC12'],
+      });
+    });
+
+    test('does not add circular dependency on bucket with extended load balancer', () => {
```

Comment:
are we checking that s3 bucket creation doesn't depend on ELB here?

Reply:
Added [test code](https://github.com/aws/aws-cdk/pull/30599/files#diff-8de1f35a0483f429c861e21e5353123bc5e610f98602b0cccc7f6f877b3a6848R982) to verify that there is no `DependsOn` on the bucket to ensure that the bucket storing the logs does not set an explicit dependency on the bucket policy.

Here is the cause for this test case to be added for reference.
https://github.com/aws/aws-cdk/issues/1633#issuecomment-1758171559
https://github.com/aws/aws-cdk/pull/27558